common/models/middleware.js

```diff
@@ -299,7 +324,11 @@ function ready(Middleware) {
     if (data.subPhase) {
       phase = phase + ':' + data.subPhase;
     }
-    return phase + '.' + data.name;
+    var index = '';
+    if (data.index) {
+      index = data.index.toString();
+    }
+    return phase + '.' + data.name + index;
```

Comment:
Please add a separator between `data.name` and `index` in order to prevent possible clashes.

Example:

```
phase: "files"
data.name: "serve-static"
index: 20

phase: "files"
data.name: "serve-static2"
index: 0
```

Suggested solution:

```js
return phase + '.' + data.name + '.' + index;
```